src/crypto/x509/x509.go

```diff
@@ -108,6 +109,12 @@ func marshalPublicKey(pub interface{}) (publicKeyBytes []byte, publicKeyAlgorith
 		}
 		publicKeyBytes, _ = pub.MarshalBinary()
 		publicKeyAlgorithm.Algorithm = scheme.Oid()
+	case *kem.PublicKey:
+		publicKeyBytes = kem.MarshalPublicKey(pub)
+		publicKeyAlgorithm.Algorithm = oidPublicKeyKEMTLS
+	case kem.PublicKey:
+		publicKeyBytes = kem.MarshalPublicKey(&pub)
+		publicKeyAlgorithm.Algorithm = oidPublicKeyKEMTLS
```

Comment:
This should be simplified, but I'm having some trouble with the conventions on when to use a pointer or struct.